### PR TITLE
fix: embedding version

### LINKED ISSUE #31: Latest changes on generating version.h breaks embedding
Hello, the latest changes (https://github.com/libinjection/libinjection/commit/bd86522711f6fd2b5f257023e19627fd6c1d5ca0) completely break embedding the files into other projects, like naxsi.

I have been using this lib for a loooong time within naxsi (an NGINX FOSS WAF) and the latest changes completely break the compatibility with other software using this library source code without actually building the .so/.a library.

i would suggest to create a default `version.h` file which gets overwritten by the configure script.
The issue with building this without actually building the library is due how NGINX plugins build system is done which does not simply allow to build the lib first and then link to it easily like any other good build system.

### PR DESCRIPTION
**Pull request by fzipi**
- remove unused version.h.in file
- define LIBINJECTION_VERSION based on PACKAGE_VERSION when used in autotools
- add comment in README on how to define the version when embedding

Fixes #31.

### CODE REVIEW

**Comment by wargio** [comment]
Tried on naxsi side and i can build it with this branch.

**Comment by fzipi** [comment]
@migueldemoura can you approve/comment on this one also?

File: src/libinjection_sqli.c [hunk 1]
```diff
@@ -14,11 +14,15 @@
 #include <assert.h>
 #include <stddef.h>
 
-#include "version.h"
 #include "libinjection.h"
 #include "libinjection_sqli.h"
 #include "libinjection_sqli_data.h"
 
+// make clang analyzer happy by defining a dummy version
```
Comment (migueldemoura) [inline src/libinjection_sqli.c:21]: Should we guard this with `__clang_analyzer__`?

Comment (fzipi) [reply src/libinjection_sqli.c:21]: Yes, makes sense.

**Review by migueldemoura — APPROVED** [review]

File: src/libinjection_sqli.c [hunk 2]
```diff
@@ -596,34 +600,33 @@ static size_t parse_operator2(struct libinjection_sqli_state * sf)
...
 }
 
+/*
+ * Ok!   "  \"   "  one backslash = escaped!
+ *       " \\"   "  two backslash = not escaped!
+ *       "\\\"   "  three backslash = escaped!
+ */
+static int is_backslash_escaped(const char* end, const char* start)
+{
+    const char* ptr;
```
Comment (migueldemoura) [inline src/libinjection_sqli.c:610]: From cppcheck: `src/libinjection_sqli.c:610 unusedVariable style Unused variable: ptr`

**Pull request merged**